lib/options.nix

```diff
@@ -313,34 +313,6 @@ rec {
         );
     };
 
-  # TODO: Deprecated 2024-09-02; remove once all internal uses are gone
-  mkPackageOption =
-    args:
-    # A default package is required
-    assert args ? default;
-    # `name` must be present if `description` is missing
-    assert (!args ? description) -> args ? name;
-    mkNullOrOption' (
-      (lib.filterAttrs (n: _: n != "name") args)
-      // {
-        type = types.package;
-        description =
-          args.description or ''
-            Which package to use for `${args.name}`.
-            Set to `null` to disable its automatic installation.
-          '';
-      }
-    );
-
-  # TODO: Deprecated 2024-09-02; remove once all internal uses are gone
-  mkPluginPackageOption =
-    name: default:
-    lib.mkOption {
-      type = types.package;
-      inherit default;
-      description = "Which package to use for the ${name} plugin.";
-    };
-
```

Comment:
Are we happy to straight up remove these without even a `throw "this function was removed, use lib.mkPackageOption instead"` style error?

It is _possible_ that end-users have started using the helpers in their own options...

Reply:
Searching github doesn't reveal much, but I don't think it'd hurt to throw the help message.

Reply:
Sure thing, done in 6c1a6182627317492ab9175e89a9afd65f99c5ff.